CI is not testing the pygmsh tests and missed an issue in #624

```
FAILED pytest/test_io_meshes.py::test_from_meshio_pygmsh[generate_pygmsh_sphere] - ValueError: operands could not be broadcast together with shapes (180,3) (179,3)
```

--- a/pytest/test_bodies.py
+++ b/pytest/test_bodies.py
@@ -157,6 +157,35 @@ def test_clipping_of_dofs(z_center, as_collection_of_meshes):
         assert len(clipped_sphere.dofs['test_dof']) == 0
 
 
+def test_complicated_clipping_of_dofs():
+    # 1 face becomes 2 faces after clipping
+    mesh = cpt.Mesh(vertices=[[0.0, 0.0, 0.5], [-0.5, 0.0, -0.5], [0.0, 0.0, -1.5], [0.0, 0.5, -0.5]], faces=[[0, 1, 2, 3]])
+    body = cpt.FloatingBody(mesh, dofs=cpt.rigid_body_dofs())
+    clipped_body = body.immersed_part()
+    assert len(clipped_body.dofs["Heave"]) == clipped_body.mesh.nb_faces
+
+
+@pytest.mark.xfail
+def test_clipping_of_dofs_with_degenerate_faces():
+    vertices = np.array([
+        [-8.00000000e+00,  1.65358984e+00, -4.99999996e-02],
+        [-8.00000000e+00,  1.65358984e+00,  5.00000003e-02],
+        [-8.00000000e+00,  1.74019238e+00, -9.99999998e-02],
+        [-8.00000000e+00,  1.74019238e+00, -1.78037182e-10],
+        [-8.00000000e+00,  1.74019238e+00,  1.00000000e-01],
+        [-8.00000000e+00,  1.82679492e+00, -5.00000002e-02],
+        [-8.00000000e+00,  1.82679492e+00,  4.99999997e-02]
+        ])
+    faces = np.array([
+        [3, 4, 6, 3],
+        [2, 0, 3, 2],
+        ])
+    mesh = cpt.Mesh(vertices, faces)
+    body = cpt.FloatingBody(mesh, dofs=cpt.rigid_body_dofs())
+    clipped_body = body.immersed_part()
+    assert len(clipped_body.dofs["Heave"]) == clipped_body.mesh.nb_faces
+
+
 def test_cropping_body_with_manual_dof():
     # https://github.com/capytaine/capytaine/issues/204
     sphere = cpt.FloatingBody(cpt.mesh_sphere())

--- a/pytest/test_io_meshes.py
+++ b/pytest/test_io_meshes.py
@@ -61,7 +61,7 @@ def test_from_meshio_pygmsh(generate_pygmsh):
     mesh = cpt.load_mesh(mesh)
     assert mesh.immersed_part().volume == pytest.approx(vol_exp, rel=1e-1)
 
-    fb = cpt.FloatingBody(mesh)
+    fb = cpt.FloatingBody(mesh.immersed_part())
     fb.add_translation_dof(name="Heave")
     test_matrix = xr.Dataset(coords={
         'omega': [np.pi],

--- a/pytest/test_meshes.py
+++ b/pytest/test_meshes.py
@@ -142,6 +142,28 @@ def test_heal_mesh_removes_degenerate_panels():
     mesh.heal_mesh()
     assert mesh.nb_faces == 1
 
+@pytest.mark.xfail
+def test_heal_mesh_with_complicated_connectivities():
+    vertices = np.array([
+        [-8.00000000e+00,  1.65358984e+00, -4.99999996e-02],
+        [-8.00000000e+00,  1.65358984e+00,  5.00000003e-02],
+        [-8.00000000e+00,  1.74019238e+00, -9.99999998e-02],
+        [-8.00000000e+00,  1.74019238e+00, -1.78037182e-10],
+        [-8.00000000e+00,  1.74019238e+00,  1.00000000e-01],
+        [-8.00000000e+00,  1.82679492e+00, -5.00000002e-02],
+        [-8.00000000e+00,  1.82679492e+00,  4.99999997e-02]
+        ])
+    faces = np.array([
+        [5, 3, 6, 5],
+        [5, 2, 3, 5],
+        [3, 4, 6, 3],
+        [3, 4, 6, 3],
+        [2, 0, 3, 2],
+        [0, 1, 3, 0],
+        [1, 4, 3, 1]
+        ])
+    mesh = cpt.Mesh(vertices, faces)
+    mesh.heal_mesh()
 
 def test_extract_one_face():
     i = 2

